consensus/qkchash/qkchash.go

```diff
@@ -2,21 +2,102 @@ package qkchash
 
 import (
 	"encoding/binary"
+	"sort"
+	"sync"
+
 	"github.com/QuarkChain/goquarkchain/cluster/config"
 	"github.com/QuarkChain/goquarkchain/consensus"
 	"github.com/QuarkChain/goquarkchain/core/state"
 	"github.com/QuarkChain/goquarkchain/core/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+const (
+	cacheEntryCnt   = 1024 * 64
+	cacheAheadRound = 64 // 64*30000
 )
 
+var (
+	EpochLength = uint64(30000) //blocks pre epoch
+)
+
+type cacheSeed struct {
+	mu     sync.Mutex
```

Comment:
It will be great if a few comments about the protection scope of the mu is added.